src/main_controller.ts

```diff
@@ -381,9 +362,36 @@ export class MainController implements vscode.Disposable {
     private setClientInfo() {
         const versionString = this.extContext.extension.packageJSON.version as string;
         const [major, minor, patch] = [...versionString.split(".").map((n) => +n), 0, 0, 0];
+        logger.debug(`Setting client info: vscode-neovim ${major}.${minor}.${patch}`);
         this.client.setClientInfo("vscode-neovim", { major, minor, patch }, "embedder", {}, {});
     }
 
+    private async setCurrentDir() {
```

Comment:
neither the linked issue, nor this PR , nor the commit message, explain why this works but setting `cwd` in the spawn() call doesn't. A docstring on this function would be helpful.

In the future, please explain the problem that is being solved by each PR, in the commit message.

Reply:
![image](https://github.com/vscode-neovim/vscode-neovim/assets/47070852/b357f83a-fc71-4018-b58c-a73ee79e7692)